src/main/java/Command.java

```diff
@@ -0,0 +1,263 @@
+        }
+        String[] modifiedCommand = removeFirst(commandArr);
+        String[] upper = removeAfterWord(modifiedCommand, "/at");
+        String[] lower = keepAfterWord(modifiedCommand, "/at");
+        LocalDateTime dateAndTime = Parser.changeDateAndTime(lower);
+        Task newTask = new Event(joinString(upper), dateAndTime);
+        return newTask;
+    }
+
+    /**
+     * Search keyword from all the Tasks' description in the TaskList given and
+     * return arraylist of task that match the keyword.
+     *
+     * @param tasks TaskList
+     * @param keyword string
+     * @return ArrayList of Task
+     */
+    public ArrayList<Task> searchKeyWord(TaskList tasks, String keyword) {
+        ArrayList<Task> tempTasks = new ArrayList<>();
+        for (Task task: tasks.tasks) {
+            if (task.description.matches("(.*)" + keyword + "(.*)")) {
+                tempTasks.add(task);
+            }
+        }
+        return tempTasks;
+    }
+
+    /**
+     * Execute Command.
+     *
+     * @param tasks TaskList
+     * @param ui Ui
+     * @param storage Storage
+     * @throws InvalidTodoDescripDukeException
+     * @throws InvalidDeadlineDescripDukeException
+     * @throws InvalidEventDescripDukeException
+     * @throws InvalidFirstDukeException
+     * @throws ParseException
+     * @throws IOException
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws InvalidTodoDescripDukeException,
+            InvalidDeadlineDescripDukeException, InvalidEventDescripDukeException,
+            InvalidFirstDukeException, ParseException, IOException {
+
+        if (commandType.equals(CommandType.PRINTALLTASKS)) {
+            printAllTask(tasks);
+        } else if (commandType.equals(CommandType.EXITDUKE)) {
+            ui.closeDuke();
+        } else if (commandType.equals(CommandType.MARKASDONE)){
+            int counter = Integer.parseInt(commandArr[1]);
+            tasks.getTask(counter - 1).markAsDone();
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.DELETETASK)){
+            int counter = Integer.parseInt(commandArr[1]);
+            deleteTask(counter - 1, tasks);
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.ADDTODO)){
+            tasks.addTask(createTodo());
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.ADDDEADLINE)) {
+            tasks.addTask(createDeadline());
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.ADDEVENT)) {
+            tasks.addTask(createEvent());
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.FINDTASK)) {
+            String keyword = joinString(removeFirst(commandArr));
+            ArrayList<Task> tempTasks = searchKeyWord(tasks, keyword);
+            printSearchedTask(tempTasks);
+        } else {}
```

Comment:
There is no coding violation here but would it better if you change the multiple else-if statements to a switch statement and handle each different case within the switch statement? This might help to improve code readability and make the code even cleaner.

Reply:
Ok, working on it